src/components/touchables/GenericTouchable.tsx

```diff
@@ -14,61 +24,51 @@ import PropTypes from 'prop-types';
  * travel outside it transits to special MOVED_OUTSIDE state. Gesture recognition
  * finishes in UNDETERMINED state.
  */
-export const TOUCHABLE_STATE = {
+export const TOUCHABLE_STATE: Record<string, TouchableStateValuesType> = {
   UNDETERMINED: 0,
   BEGAN: 1,
   MOVED_OUTSIDE: 2,
 };
 
-const PublicPropTypes = {
+type TouchableStateValuesType = 0 | 1 | 2;
```

Comment:
If it works, I am ok with it.

Reply:
I can change it to: 
```ts
export const TOUCHABLE_STATE = {
  UNDETERMINED: 0,
  BEGAN: 1,
  MOVED_OUTSIDE: 2,
} as const;

type TouchableStateType = typeof TOUCHABLE_STATE[keyof typeof TOUCHABLE_STATE];
```

It will then follow the `State` implementation.

Reply:
Definitely looks more maintainable IMO.